src/Buckle/Buckle/CodeAnalysis/Binding/Binder.cs

```diff
@@ -848,8 +844,8 @@ private struct BinderState {
     }
 
     private BoundStatement BindWhileStatement(WhileStatement statement, bool insideInline = false) {
-        var conditionValue = RemoveNullability(statement.condition);
-        var condition = BindCast(conditionValue, BoundTypeClause.NullableBool);
+        var condition = BindDoubleCastRemovingNullability(
```

Comment:
Rewrote all the conditional bindings to factor out duplicate code into BindDoubleCastRemovingNullability